In [1]:
import pandas as pd

In [2]:
# hpar: house-prices-advanced-regression
hpar_train_path = '../../data/data_small/house-prices-advanced-regression-techniques/train.csv'
hpar_test_path = '../../data/data_small/house-prices-advanced-regression-techniques/test.csv'
hpar_df_train = pd.read_csv(hpar_train_path)
hpar_df_test = pd.read_csv(hpar_test_path)
hpar_df = pd.concat([hpar_df_train, hpar_df_test])

hpar_df

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0        1          60       RL         65.0     8450   Pave   NaN      Reg   
1        2          20       RL         80.0     9600   Pave   NaN      Reg   
2        3          60       RL         68.0    11250   Pave   NaN      IR1   
3        4          70       RL         60.0     9550   Pave   NaN      IR1   
4        5          60       RL         84.0    14260   Pave   NaN      IR1   
...    ...         ...      ...          ...      ...    ...   ...      ...   
1454  2915         160       RM         21.0     1936   Pave   NaN      Reg   
1455  2916         160       RM         21.0     1894   Pave   NaN      Reg   
1456  2917          20       RL        160.0    20000   Pave   NaN      Reg   
1457  2918          85       RL         62.0    10441   Pave   NaN      Reg   
1458  2919          60       RL         74.0     9627   Pave   NaN      Reg   

     LandContour Utilities  ... PoolArea PoolQC  Fence MiscFeature MiscVal  \
0            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
2            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
3            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
4            Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
...          ...       ...  ...      ...    ...    ...         ...     ...   
1454         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1455         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1456         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   
1457         Lvl    AllPub  ...        0    NaN  MnPrv        Shed     700   
1458         Lvl    AllPub  ...        0    NaN    NaN         NaN       0   

     MoSold YrSold  SaleType  SaleCondition  SalePrice  
0         2   2008        WD         Normal   208500.0  
1         5   2007        WD         Normal   181500.0  
2         9   2008        WD         Normal   223500.0  
3         2   2006        WD        Abnorml   140000.0  
4        12   2008        WD         Normal   250000.0  
...     ...    ...       ...            ...        ...  
1454      6   2006        WD         Normal        NaN  
1455      4   2006        WD        Abnorml        NaN  
1456      9   2006        WD        Abnorml        NaN  
1457      7   2006        WD         Normal        NaN  
1458     11   2006        WD         Normal        NaN  

[2919 rows x 81 columns]

# House Prices - Advanced Regression Techniques Dataset
https://www.kaggle.com/c/house-prices-advanced-regression-techniques

## Descripción del dataset

El House Prices: Advanced Regression Techniques es un conjunto de datos ampliamente utilizado en la comunidad de ciencia de datos, disponible en Kaggle (House Prices: Advanced Regression Techniques). Recopila información detallada sobre ventas de casas residenciales en Ames, Iowa, entre 2006 y 2010. El dataset 
contiene 2919 filas y 80 variables originalmente viene dividido en instancias de training y testing. Cabe aclarar que el subconjunto para testing no incluye la variable de "SalePrice" ya que este dataset es una competencia actual en Kaggle.

Cada fila representa una casa vendida, con atributos que describen características físicas (e.g., tamaño, número de habitaciones), calidad (e.g., materiales, condición), ubicación (e.g., vecindario) y el precio de venta (SalePrice). Este dataset es ideal para explorar factores que influyen en los precios inmobiliarios, desarrollar modelos predictivos y analizar patrones en el mercado de viviendas, ofreciendo oportunidades para aplicar técnicas de regresión, visualización y segmentación en un contexto comercial y analítico relevante.

## Posibles Análisis y Atributos Relevantes

A continuación, detallo cuatro posibles análisis que se pueden realizar con este dataset, los atributos relevantes para cada uno y su importancia en contextos comerciales y analíticos:

1. Predicción del Precio de Venta de Casas
   
   Variables potencialmente interesantes:
   1. **SalePrice:** Variable objetivo numérica, el precio de venta en dólares, a predecir. 
   2. **GrLivArea:** Área habitable sobre el suelo (en pies cuadrados), un predictor clave del tamaño y valor.
   3. **OverallQual**: Calidad general del material y acabado (escala 1-10), influye fuertemente en el precio.
   4. **Neighborhood**: Vecindario, captura efectos de ubicación en el valor de la propiedad.  
   5. **YearBuilt**: Año de construcción, refleja la antigüedad y estilo de la casa.

2. Análisis de Factores que Impactan el Precio por Vecindario
   
   Variables potencialmente interesantes:
   1. **SalePrice**: Precio de venta, base para comparación entre vecindarios
   2. **Neighborhood**: Variable categórica que define la zona, clave para análisis espacial.
   3. **LotArea**: Tamaño del lote (en pies cuadrados), afecta el valor según la ubicación.
   4. **TotalBsmtSF**: Área total del sótano, indica espacio adicional por vecindario.
   5. **GarageArea**: Área del garaje, un diferenciador en ciertas zonas.  

3. Segmentación de Casas por Características Físicas y de Calidad
   
   Variables potencialmente interesantes:
   1. **GrLivArea**: Área habitable, define el tamaño de la casa
   2. **OverallQual**: Calidad general, segmenta por nivel de lujo o estándar.
   3. **TotRmsAbvGrd**: Número total de habitaciones sobre el suelo, indica capacidad familiar.
   4. **BldgType**: Tipo de edificio (e.g., unifamiliar, dúplex), útil para clasificar propiedades.

4. Análisis de Tendencias Temporales en Precios de Vivienda
   
   Variables potencialmente interesantes: 
   1. **SalePrice**: Precio de venta, base para analizar tendencias.  
   2. **YrSold**: Año de venta, permite estudiar cambios anuales.  
   3. **YearBuilt**: Año de construcción, mide el impacto de la antigüedad.  
   4. **MoSold**: Mes de venta, captura estacionalidad en precios.
